src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -765,7 +765,7 @@ private void parsePublicKey(final List<Credential> credentials, final XMLStreamR
                 if (algorithm == null) {
                     throw ElytronMessages.log.fileSystemRealmMissingAttribute("algorithm", path, streamReader.getLocation().getLineNumber(), name);
                 }
-                if (! format.equals("pkcs8")) {
+                if (format == null || !format.equals("pkcs8")) {
```

Comment:
If the format isn't given, surely we want to assume "X.509" (else require it).  Also - oops - PKCS8EncodedKeySpec only works on private keys; we should be using X509EncodedKeySpec instead here!